packages/typegpu/src/std/numeric.ts

```diff
@@ -31,92 +37,192 @@ export function isNumeric(element: Snippet) {
   );
 }
 
+function cpuAdd(lhs: number, rhs: number): number; // default addition
+function cpuAdd<T extends Vec>(lhs: number, rhs: T): T; // mixed addition
+function cpuAdd<T extends Vec>(lhs: T, rhs: number): T; // mixed addition
+function cpuAdd<T extends Vec | Mat>(lhs: T, rhs: T): T; // component-wise addition
+function cpuAdd<
+  // union overload for internal usage
```

Comment:
It's for external usage as well, when somebody wants to pass in a union of values